dms_field/tests/test_dms_field.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests import SavepointCase
+
+
+class TestDmsField(SavepointCase):
+    def setUp(self):
+        super().setUp()
+        self.storage = self.env["dms.storage"].create(
+            {
+                "name": "DMS Storage",
+                "model_ids": [(4, self.env.ref("base.model_res_partner").id)],
+            }
+        )
+        self.partner = self.env["res.partner"].create({"name": "DEMO Partner"})
+
+    def test_check_constrain_multi_directory(self):
+        self.env["dms.directory"].create(
+            {
+                "root_storage_id": self.storage.id,
+                "is_root_directory": True,
+                "name": "First Directory",
+                "res_model": self.partner._name,
+                "res_id": self.partner.id,
+            }
+        )
```

Comment:
```suggestion
    def _create_directory_vals(self, record):
        return {
            "root_storage_id": self.storage.id,
            "is_root_directory": True,
            "name": record.display_name,
            "res_model": record._name,
            "res_id": record.id,
        }
    
    def test_check_constrain_multi_directory(self):
        self.env["dms.directory"].create(self._create_directory_vals(self.partner))
```
Something like these help to reduce code (re-use `_create_directory_vals` function or similar).